### DISCUSSION TITLE
Strange Bug on boot after an update (back to work, but the exception didnt help)

### DISCUSSION THREAD

**Discussion by patrickmaciel** [livewire/livewire#4508] [7. Bugs (archived)]
Hi guys.

- Livewire 2.8.2
- Laravel 8.78.1
- PHP 7.4.19

I have a specific problem that happens when I have a double negative value in the page in resume:

- when I have positive values on page, everything is ok
- when I have a negative value, the exception happens

Example: in the image below, when I update the 2 inputs values with a positive value, everythings works great, but when the calculations return a negative value, the exception appear.

![image](https://user-images.githubusercontent.com/671670/148965138-64074458-1ff4-4966-9b7c-cf179272c73d.png)

Code related to the screenshot

```blade.php
<ul role="list" class="grid grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-3">
    @forelse ($precatorioPropostasIndividuais as $index => $ppi)
        <li wire:key='precatorioPropostasIndividuais.{{ $index }}'
            class="p-4 bg-gray-100 rounded-md border-2 hover:bg-gray-300 font-medium">
            @if ($ppi->precatorio_advogado_id == null)
                CREDOR - {{ $precatorio->nome }}
            @else
                {{ $ppi->precatorioAdvogado->escritorio ? 'ESCRITÓRIO' : 'ADVOGADO' }}
                - {{ $ppi->precatorioAdvogado->nome }}
            @endif
            <x-form.col class='mt-2'>
                <x-input.label
                    for='precatorioPropostasIndividuais.{{ $index }}.proposta_minima'
                    label='Valor máximo de proposta' />
                <x-input.text
                    wire:model.debounce.lazy='precatorioPropostasIndividuais.{{ $index }}.proposta_minima'
                    prepend='R$' type='number' step='0.000000000000001'
                    min='0.000000000000000'
                    name='precatorioPropostasIndividuais.{{ $index }}.proposta_minima'
                    readonly disabled />
            </x-form.col>
            <x-form.col class='mt-2'>
                <x-input.label
                    for='precatorioPropostasIndividuais.{{ $index }}.proposta'
                    label='Proposta' />
                <x-input.text
                    wire:model.debounce.lazy='precatorioPropostasIndividuais.{{ $index }}.proposta'
                    prepend='R$' type='number' step='0.000000000000001'
                    min='0.000000000000000'
                    name='precatorioPropostasIndividuais.{{ $index }}.proposta' />
            </x-form.col>
            <x-form.col class='mt-2'>
                <x-input.label
                    for='precatorioPropostasIndividuais.{{ $index }}.desagio'
                    label='Deságio' />
                <x-input.text
                    wire:model.debounce.lazy='precatorioPropostasIndividuais.{{ $index }}.desagio'
                    prepend='%' type='number' step='0.000000000000001'
                    min='0.000000000000000'
                    name='precatorioPropostasIndividuais.{{ $index }}.desagio' />
            </x-form.col>
        </li>
    @empty
        <li>Nenhum advogado foi vinculado a esse precatório</li>
    @endforelse
</ul>
```

**Livewire encountered corrupt data when trying to hydrate the [pages.precatorios.visualizacao] component. Ensure that the [name, id, data] of the Livewire component wasn't tampered with between requests.** `vendor\livewire\livewire\src\HydrationMiddleware\SecureHydrationWithChecksum.php:21`

I record a video to try to explain the situation. Sorry for my bad English.
I'll do not post any code here because livewire does not call any of my methods, it's a core method.

https://www.loom.com/share/e2c01295171a41df87fa9be0a6b8c08f

**Comment by patrickmaciel** [comment]
Stack Trace

![image](https://user-images.githubusercontent.com/671670/148988620-6b42b5de-3072-4be2-ae51-73d29ebfac91.png)

**Comment by patrickmaciel** [comment]
I put debug functions in all lifecycle methods, and after update the input on the example above only the method `boot` was called.

**Comment by patrickmaciel** [comment]
I put some debugs, refresh, clear the cache and after some hours, back to work with no reason.
That exception ruined my day, omg.

**Comment by joshhanley** [comment]
@patrickmaciel hmm yeah that can be a tough one to debug. If I remember rightly, I think if there is a `-0` or something like that in the DB or input it can potentially cause that issue. See this discussion https://github.com/livewire/livewire/discussions/3053#discussioncomment-1115663

Hope this helps!
